src/main/java/org/opentripplanner/netex/mapping/TripPatternMapper.java

```diff
@@ -157,9 +157,7 @@ class TripPatternMapper {
     }
   }
 
-  TripPatternMapperResult mapTripPattern(JourneyPattern_VersionStructure journeyPattern) {
-    // Make sure the result is clean, by creating a new object.
-    result = new TripPatternMapperResult();
+  Optional<TripPatternMapperResult> mapTripPattern(JourneyPattern_VersionStructure journeyPattern) {
```

Comment:
Observation: The original result class did handle the empty case, so this is weakening the existing functionality. The good thing is that a it is a bit less code to maintain, but since this does not reduce the code in the caller - it complicates it, than gain is minimal. I do not suggest to change it, but just wanted to mention it.

Reply:
I wanted to make it explicit that if we return a result then it will have a single TripPattern. I could have added an optional/nullable `tripPattern` field to the result class instead but that wouldn't really be simpler and i think this better expresses the actual behaviour of the mapTripPattern method.

Now the caller has to handle the case where the result is empty, it's a bit more complicated at the call site but IMO it's worth it.

Reply:
My point is that you are wrapping a result wrapper with another result wrapper - Optional and TripPatternMapperResult have some of the same responsibilities. You should not wrap the field inside `TripPatternMapperResult` with optional. We are return a collection here, not a single instance - you could provide a `isEmpty()` method to make it explicit, but it was not needed; hence not added. The pattern we follow in OTP for collections is to return empty collections, not `null` or Optional.

Reply:
Another thing is that we avoid using records with arrays and collections. Arrays can not be protected, while collections are hard to protect - classes are simpler/less error prune. Even if the `TripPatternMapperResult` is just a DTO here, people have a tendency to copy code/and ways of doing things - so I want it to be clean everywhere - even if it in this case would be ok.

Reply:
> My point is that you are wrapping a result wrapper with another result wrapper - Optional and TripPatternMapperResult have some of the same responsibilities.

In this PR my idea is to change the responsibilities so that the result class is only responsible for representing an actual result. The Optional is responsible for representing presence/absence of a result. I think this is a pretty common pattern. And I think it is a good pattern since the result class would only have to concern itself with a single responsibility.

> We are return a collection here, not a single instance - you could provide a `isEmpty()` or `ifPresent()` method to make it explicit, but it was not needed; hence not added. The pattern we follow in OTP for collections is to return empty collections, not `null` or Optional.

The thing is that I want the result to have a single instance of a TripPattern instead of a collection of TripPatterns. Because I want the caller to be able to see what the resulting id is and I wanted it to be obvous that the issue for duplicated ids is only reported once.

In this codebase, would it be more idiomatic to put a `List<TripPattern>` in the result and have it be implicit that this is either empty or has a single entry?

Reply:
> In this codebase, would it be more idiomatic to put a List<TripPattern> in the result and have it be implicit that this is either empty or has a single entry?

Of cause not. The `TripPatternMapperResult` has a single-responsibility - to carry the result of one method.  The result has many different states, which you toke just one out. 

But we can agree to disagree - that is fine. I will approve the PR.